playwright/all_tests/assignment/assignmentHelpers.ts

```diff
@@ -74,127 +122,111 @@ export async function updateSukoAssignmentForm({
+  await selectDropdownOption(page, 'aineKoodiArvo', '1')
 
-  await fillLaajalainenOsaaminen(page)
-
-  await page.getByTestId('nameFi').fill(nameTextFi)
-  await page.getByTestId('editor-content-fi').locator('div[contenteditable="true"]').fill(contentTextFi)
-
-  await page.getByTestId('tab-sv').click()
+  await fillLukuvuosi(page)
 
-  await page.getByTestId('nameSv').fill(nameTextSv)
-  await page.getByTestId('editor-content-sv').locator('div[contenteditable="true"]').fill(contentTextSv)
+  // Eettisyys ja ympäristöosaaminen
+  await fillMultiselectDropdownOption(page, 'laajaalainenOsaaminenKoodiArvos', ['05'])
+
+  await fillFinAndSvTextFields(
+    page,
+    { fi: contentTextFi, sv: contentTextSv },
+    { fi: nameTextFi, sv: nameTextSv },
+    { fi: instructionTextFi, sv: instructionTextSv }
+  )
 }
 
 export async function fillPuhviAssignmentForm({
   page,
   contentTextFi,
   contentTextSv,
   nameTextFi,
-  nameTextSv
-}: {
-  page: Page
-  nameTextFi: string
-  nameTextSv: string
-  contentTextFi: string
-  contentTextSv: string
-}) {
+  nameTextSv,
+  instructionTextFi,
+  instructionTextSv
+}: FillAssignmentForm) {
   // Esiintymistaidot
   await page.getByTestId('assignmentTypeRadio-002').click()
+  await fillLukuvuosi(page)
+  await fillMultiselectDropdownOption(page, 'laajaalainenOsaaminenKoodiArvos', ['05'])
+
+  await fillFinAndSvTextFields(
+    page,
+    { fi: contentTextFi, sv: contentTextSv },
+    { fi: nameTextFi, sv: nameTextSv },
+    { fi: instructionTextFi, sv: instructionTextSv }
+  )
+}
 
-  await page.getByTestId('lukuvuosiKoodiArvos').click()
-  // 2020-2021
-  await page.getByTestId('lukuvuosiKoodiArvos-option-20202021').click()
-  // Close dropdown onBlur
-  await page.getByTestId('lukuvuosiKoodiArvos-label').click()
+export async function updateLdAndPuhviAssignment(page: Page) {
+  await page.getByTestId('laajaalainenOsaaminenKoodiArvos-input').click()
+  await page.getByTestId('laajaalainenOsaaminenKoodiArvos-option-02').click()
+  await page.getByTestId('laajaalainenOsaaminenKoodiArvos-multi-select-ready-button').click()
 
-  await fillLaajalainenOsaaminen(page)
+  await fillFinAndSvTextFields(
+    page,
+    { fi: 'Testi sisältö muokattu', sv: 'Testa innehåll muokattu' },
+    { fi: 'Testi tehtävä muokattu', sv: 'Testuppgifter muokattu' },
+    { fi: 'Testi ohjeet muokattu', sv: 'Testa instruktioner muokattu' }
+  )
 
-  await page.getByTestId('nameFi').fill(nameTextFi)
-  await page.getByTestId('editor-content-fi').locator('div[contenteditable="true"]').fill(contentTextFi)
+  await page.getByTestId('form-submit').click()
 
-  await page.getByTestId('tab-sv').click()
+  await page.getByTestId('assignment-header').isVisible()
+  const updatedAssignmentHeader = page.getByTestId('assignment-header')
+  await expect(updatedAssignmentHeader).toHaveText('Testi tehtävä muokattu')
+  await page.getByText('Testi sisältö muokattu', { exact: true }).isVisible()
```

Comment:
Tää rivi taitaa kyllä jotenkuten toimia, mutta käyttäisin getByTestId ja toHaveText, koska se kertoo lukijalle paremmin, mitä tässä testataan. Ja sama noihin alempiin samanlaisiin riveihin.

```suggestion
  await expect(page.getByTestId('editor-content-fi')).toHaveText('Testi sisältö muokattu'))
```